workflow/workflow.go

```diff
@@ -34,10 +37,28 @@ type Cache struct {
 	fOpts file.Options
 
 	Workflows map[string]Workflow // the key is the filename for the workflow
+	mutex     sync.Mutex          // used to update changes to any workflows
+}
+
+// InitRefresh will initialize a go routine to refresh the workflow(s) based on a duration
+func (c *Cache) InitRefresh(dur string) {
+	d, err := time.ParseDuration(dur)
+	if err != nil {
+		log.Fatal("could not parse workflow refresh duration")
+	}
+	// starts a looping routine to update workflow changes after a time duration
+	go func(d time.Duration) {
+		for now := range time.Tick(d) {
+			fmt.Println("refreshing workflows", now)
+			c.mutex.Lock()
+			c.Refresh()
+			c.mutex.Unlock()
+		}
+	}(d)
 }
 
 // New returns a Cache used to manage auto updating a workflow
-func New(path string, opts *file.Options) (*Cache, error) {
+func New(path, dur string, opts *file.Options) (*Cache, error) {
```

Comment:
Instead of modifying new, maybe we should add a new method that starts auto-updating the workflows. 
``` go 
func (c *Cache) AutoUpdate(interval time.Duration) (updateChan chan <- struct{}) 
```

Reply:
I'm not sure how that works, but I'm open to ideas. It just makes sense to me to have an update duration when you create a new cache.